main.py

```diff
@@ -46,32 +52,71 @@ def read_files(args):
     # Create data instance
     data = dataset.Data()
 
-    # Read Films file
-    lines=films_file.readlines()
+    # Read files
+    films_lines=films_file.readlines()
+    collectors_lines=collectors_file.readlines()
+    director_id=3000
+    actor_id=2000
+
+    # Firstly, read informations in FILMS file
+    for i in xrange(len(films_lines)):
+        line=films_lines[i].split(' % ')
 
-    for i in xrange(len(lines)):
-        line=lines[i].split(' % ')
-        # Create dictionary for movies
+        #Send movie informations to dataset and create a movie
         movie = dataset.Movie(
             ID=line[0],
             title=line[1],
             year=line[2],
             genre=line[4],
             director=line[5],
-            rating=line[6]
+            rating=line[6],
         )
-        # Divide actors(line[3]=actors) incording to "," and add into actor_list
-        movie.actors = Set(line[3].split(', '))
+        movie.directors.add(line[5])
+        movie.actors=Set(line[3].split(', '))
+
+        #Send informations of every director to Director class in dataset
+        for director in movie.directors:
+            director=dataset.Director(
+                ID=director_id,
+                name=director
+            )
+            director_id+=1
+            # Add director into directors set in dataset
+            data.directors.add(director)
+
+        #Send every actor to Actor class in dataset
+        for actor in movie.actors:
+            #Create actor
+            actor=dataset.Actor(
+                ID=actor_id,
+                name=actor
+            )
+            actor_id+=1
+            # Add actor into actors set in data class
+            data.actors.add(actor)
+
+        #Add movie into movies set in dataset

```

Comment:
#Add movie into movies set in dataset
         data.movies.add(movie)
bu iki satırı direk movie'nin tamamlandığı yerin altına koyuyorum.